lib/internal/cluster/round_robin_handle.js

```diff
@@ -10,115 +10,100 @@ const net = require('net');
 const { sendHelper } = require('internal/cluster/utils');
 const { constants } = internalBinding('tcp_wrap');
 
-module.exports = RoundRobinHandle;
-
-function RoundRobinHandle(key, address, port, addressType, fd, flags) {
-  this.key = key;
-  this.all = new Map();
-  this.free = [];
-  this.handles = [];
-  this.handle = null;
-  this.server = net.createServer(assert.fail);
-
-  if (fd >= 0)
-    this.server.listen({ fd });
-  else if (port >= 0) {
-    this.server.listen({
-      port,
-      host: address,
-      // Currently, net module only supports `ipv6Only` option in `flags`.
-      ipv6Only: Boolean(flags & constants.UV_TCP_IPV6ONLY),
+class RoundRobinHandle {
+  constructor(key, address, { port, fd, flags }) {
+    this.key = key;
+    this.all = new Map();
+    this.free = new Map();
```

Comment:
I think it's best to keep this as an array since an array is most likely designed better for the purposes of a queue than a `Map` is. The only real benefit of the `Map` is faster lookups, but we shouldn't really optimize that over faster/easier `shift()`/`push()` because workers are not likely to be added/removed as often as workers being selected for requests.

Reply:
True, but shifting logic is at the time of distributing the load and that is a single operation that can be implemented using keys iterator on map is still an O(1) operation and getting the element and deleting that from the Hash is same on average. Reimplemented to delete the worker from free pool.